test/async-hooks/test-pipeconnectwrap.js

```diff
@@ -7,9 +7,13 @@ const initHooks = require('./init-hooks');
 const { checkInvocations } = require('./hook-checks');
 
 const net = require('net');
+const fs = require('fs');
 
+// Verify `tmpdir` exists for platforms that need it for `common.PIPE`.
 const tmpdir = require('../common/tmpdir');
-tmpdir.refresh();
+if (!fs.existsSync(tmpdir.path)) {
```

Comment:
I'd prefer we continue rimraf'ing the tmp directory so that there isn't any leftover cruft that might somehow interfere with the test. (Pipe already exists or something? I have no idea. Really, it just seems like a bit of a test/code smell to me to do it this way when `tmpdir.refresh()` works for all other tests.) Would it fix it to move the require for `tmpdir` and the invocation of `tmpdir.refresh()` to start on line 4, so that it all happens before any of the async-hooks stuff? Or does it still mess up the state there?

As a last resort, I guess I'd prefer a config object be passed to `tmpdir.refresh()` with a `spawn: false` option or something like that? 

(If we don't end up changing this code to something like any of the above, can you put a comment here explaining why we're not using `tmpdir.refresh()` so someone (like me) doesn't come along in three months and change this back to `tmpdir.refresh()`?)

Reply:
> Would it fix it to move the require for `tmpdir` and the invocation of `tmpdir.refresh()` to start on line 4, so that it all happens before any of the async-hooks stuff? Or does it still mess up the state there?

I fiddled with location and such but could find a way to keep the test passing without significant changes.

> As a last resort, I guess I'd prefer a config object be passed to `tmpdir.refresh()` with a `spawn: false` option or something like that, and then `tmpdir.refresh()` can check that before invoking `spawnSync()`?

I went with a variation on this.